ICAFIX/ReApplyFixMultiRunPipeline.sh

```diff
@@ -73,6 +73,13 @@ PARAMETERs are [ ] = optional; < > = user supplied value
      If TRUE, deletes both the concatenated high-pass filtered and non-filtered timeseries files
      that are prerequisites to FIX cleaning.
      (The concatenated, hpXX_clean timeseries files are preserved for use in downstream scripts).
+   [--vol-wisharts=<integer>] - Number of wisharts to fit to volume data in
+     icaDim, default ${G_DEFAULT_VOLWISHARTS}
+   [--cifti-wisharts=<integer>] - Number of wisharts to fit to cifti data in
+     icaDim, default ${G_DEFAULT_CIFTIWISHARTS}
+   [--icadim-mode=<"default" or "melodiclike">] - Choose how to run icaDim:
+     "default" - start with a VN dimensionality of 1 and rerun until convergence
```

Comment:
Can we elaborate slightly on what "VN dimensionality" is?

Reply:
What is "variance normalization dimensionality"?

Reply:
i.e., "variance normalization" is division by an estimate of the std. dev.  In what sense does that have a "dimensionality"?

Reply:
I would guess it is the number of PCA components to assume are "signal" for the first iteration, treating the rest as the noise estimate.  I didn't bother trying to figure out how icaDim works.  @glasserm ?

Reply:
VN=1 means divide by the timeseries standard deviation (VN=0 likely means don't do variance normalization at all).  VN=#TimePoints/2 means use the last half of the eigen spectrum to estimate the unstructured noise.  Usually the VN is adjusted dynamically with the icaDim fitting (variance normalizing with the eigenvalues smaller than the fit dimensionality.  This will tend to converge, so VN=1 is fine to start with; however with legacy data it converges to too small a number and hence we stick with a single iteration of fitting and VN=#TimePoints/2.

Reply:
It would be really helpful if, as part of this PR, we added some usage/guidance/help info to the top of `icaDim.m`.  It's pretty much impossible to understand what exactly MR-FIX is doing without being able to understand the purpose/functionality of `icaDim.m`.

Reply:
I have taken a stab at describing icaDim.

Reply:
In the info for the VN parameter in `icaDim.m`, would the following addition be accurate:
```
0: No variance normalization
1: Use standard deviation of entire timeseries
> 1: Use that many time points from the tail of the eigen spectrum to estimate the standard deviation of the unstructured noise
```
Or, is > 1 specifying the opposite, namely the number of tp's at the front of eigen spectrum to _exclude_ from the eigen spectrum for estimating the noise?

Reply:
I believe it is the opposite:

https://github.com/Washington-University/HCPpipelines/blob/29a6e7b19d598a6a7a699010b7eaacaebfb36cb0/global/matlab/icaDim/icaDim.m#L94

Also, it updates each iteration based on the dimensionality estimate (unless the argument is zero, where it skips normalization regardless).

Reply:
I'm not following the code.  How is the estimate of the unstructured noise derived from the signal components of the SVD?  I thought the unstructured noise is represented in the tail of the eigen spectrum?

Reply:
That is why the signal components are excluded: it reconstructs just the noise components into what should (in theory) be a timeseries of images containing only noise, and then takes the standard deviation of that.

Reply:
To simplify as pseudocode (with some liberties):
```
[u s v] = svd(cov(data)); %strongest components (signal) first
noiseseries = u(numsignal:end) * s(numsignal:end) * v(numsignal:end);
noisestd = stdev(noiseseries);
```
So technically it is using the ICA dimensionality estimate on a PCA decomposition.

Reply:
But that doesn't seem to be what the code is doing.
We have this:
```
[u,EigS,v]=nets_svds(data',0); %Eigenvalues of data
Out.DOF=sum(diag(EigS)>(std(diag(EigS))*0.1)); %Compute degrees of freedom
```
and then this:
`noise_unst = (u(:,Out.VNDIM(c):Out.DOF)*EigS(Out.VNDIM(c):Out.DOF,Out.VNDIM(c):Out.DOF)*v(:,Out.VNDIM(c):Out.DOF)')';`

`Out.DOF` is the equivalent of `numsignal` in your pseudocode.  But that is getting used as the end of the range of SVD modes to include, not the beginning.
i.e., what it appears to be doing is:
`noiseseries = u(1:numsignal) * s(1:numsignal) * v(1:numsignal);`

Reply:
No, DOF is not the number of signal components, it is only there because there may be some missing degrees of independence from prior processing.  `c` is the current iteration, and Out.VNDIM is the history of what each iteration estimated for the number of signal components.

Reply:
Ok, I think two different comment would be helpful:
1. Something prior to the `Out.DOF` calculation explaining what that variable represents, and why we are only including eigenvalues > 10% of the eigenvalue std as eligible "degrees of freedom" (rather than the number of total time points).  The fact that that line is selecting just the highest subset of eigenvalues made me think that it was some algorithm for selecting the "signal" components.
2. Something (maybe right in the top usage/help info) that explains more fully what "variance normalization dimensionality" represents.  If I'm understanding correctly, it is basically synonymous with "estimated or assumed number of ICA signal components"?

Reply:
The VN argument is the (signal) dimensionality used during the variance normalization of the first iteration, which is exactly how I read the current explanation.  All other iterations use the estimate produced by the previous iteration.  Other than changing the VN mode, which is now also in the explanation, it does nothing else (the weird conditional that sets `stabCount` is dead code, I have now deleted it).

Reply:
Generic "dimensionality" in the help is ambiguous.  Why can't we just make the connection explicit that "variance normalization dimensionality" is referring to the assumed (first iteration) or estimated (other iterations) _number of ICA signal components_?

Reply:
This guess of the dimensionality is only used for the variance normalization (and only in the first iteration), nothing else.  That is the only change between loop iterations, as far as I can tell.  The actual estimation code (`FitWishart`) doesn't take a dimensionality guess as input.

I would like to point out, that the intial PR treated icaDim as a black box, with no edits whatsoever (and I took a brief look at the icaDim code and decided I'd prefer not to try to understand it, if possible).  You seem to want to instead make this PR into a marathon to document the entirety of icaDim as pseudocode in the icaDim help info.  Please try to show some restraint.

Reply:
What I want to do is try to give people that come to `icaDim.m` from `functionhighpass` some basic sense of its purpose and what it is doing.  The additions you have made are very helpful in that regard.  I feel we are basically there if we just make the connection for people that all this talk of dimensionality is referring to the number of _signal components_.  I don't understand the reluctance to add that final piece of the puzzle to the icaDim help info given how helpful it would be.

Reply:
Do what you wish to wrap this up.  Thanks for all the additions.